core/services/telemetry/manager_test.go

```diff
@@ -329,7 +329,9 @@ func TestLegacyMode(t *testing.T) {
 	e2.SendLog([]byte("endpoint-2-message-1"))
 	e2.SendLog([]byte("endpoint-2-message-2"))
 	e2.SendLog([]byte("endpoint-2-message-3"))
-	require.Len(t, clientSent, 6)
+	if len(clientSent) != 6 {
```

Comment:
are we sure the linter isn't being dumb here? i've used pattern and seen it fail without a panic. i can't imagine a world were devs use this alternative here rather than assert/require.Len and therefore this will be endless toil at no gain

Reply:
Well there is an argument for readability, but yeah I suspect the linter's static analysis doesn't understand it is equivalent. We could comment them with `//nolint` instead :shrug: